addons/website/static/src/js/content/snippets.animation.js

```diff
@@ -1028,6 +1028,90 @@ registry.HeaderHamburgerFull = publicWidget.Widget.extend({
+
+    /**
+     * @override
+     */
+    start() {
+        this.$target.on('scroll.bottom_fixed_element', _.debounce(this._hideBottomFixedElements.bind(this), 100));
+        $(window).on('resize.bottom_fixed_element', _.debounce(this._hideBottomFixedElements.bind(this), 100));
+        return this._super(...arguments);
+    },
+    /**
+     * @override
+     */
+    destroy() {
+        this._super(...arguments);
+        this.$target.off('.bottom_fixed_element');
+    },
+
+    //--------------------------------------------------------------------------
+    // Handlers
+    //--------------------------------------------------------------------------
+
+    /**
+     * Hide the element that is fixed at the bottom of the screen if the scroll
+     * reaches the bottom of the page and if this element hides a button.
+     *
+     * @private
+     */
+    _hideBottomFixedElements() {
+        const self = this;
+        const $bottomFixedElements = $('.o_bottom_fixed_element');
+        if ($bottomFixedElements.length) {
+            if ((this.$target[0].offsetHeight + this.$target[0].scrollTop) >= this.$target[0].scrollHeight) {
+                const $buttons = this.$target.find('.btn');
+                _.each($bottomFixedElements, bottomFixedElement => {
+                    $(bottomFixedElement).css('transform', '');
+                    _.each($buttons, button => {
+                        if (self._isColliding($(button), $(bottomFixedElement))) {
+                            $(bottomFixedElement).css('transform', 'translate(0, 100%)');
+                        }
+                    });
+                });
+            } else {
+                $bottomFixedElements.css('transform', '');
+            }
+        }
+    },
+
+    //--------------------------------------------------------------------------
+    // Private
+    //--------------------------------------------------------------------------
+
+    /**
+     * Detects if elements are colliding
+     *
+     * @private
+     */
+    _isColliding($button, $bottomFixedElement) {
+        const buttonPos = this._getSizeAndPosition($button);
+        const bottomFixedElementPos = this._getSizeAndPosition($bottomFixedElement);
+        return !(buttonPos.distanceFromTop < bottomFixedElementPos.offset.top
+            || buttonPos.offset.top > bottomFixedElementPos.distanceFromTop
+            || buttonPos.distanceFromLeft < bottomFixedElementPos.offset.left
+            || buttonPos.offset.left > bottomFixedElementPos.distanceFromLeft);
+    },
+    /**
+     * @private
+     */
+    _getSizeAndPosition($el) {
+        const offset = $el.offset();
+        const height = $el.outerHeight( true );
+        const width = $el.outerWidth( true );
+        return {
+            offset: offset,
+            height: height,
+            width: width,
+            distanceFromTop: offset.top + height,
+            distanceFromLeft: offset.left + width,
+        };
+    },
```

Comment:
`getBoundingClientRect` should make this whole lot way smaller.

Reply:
I moved the function is web.utils as we might need it elsewhere